metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -214,6 +217,7 @@ class MetalsLspService(
   private val symbolDocs = new Docstrings(definitionIndex)
   var bspSession: Option[BspSession] =
     Option.empty[BspSession]
+  val isConnecting = new AtomicBoolean(false)
```

Comment:
Wouldn't it be enough to just use `isConnecting` and ignore any bsp files created during the connection?

Reply:
Yeah, I would try to simplify it a bit

Reply:
Okay, done, though I turned out very similar.

Reply:
Thanks!